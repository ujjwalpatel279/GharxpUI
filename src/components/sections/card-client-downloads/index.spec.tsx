import React from 'react';
import { render, cleanup } from '@testing-library/react';

import { CardClientDownloads } from './index';

const handleRefreshDocuments = jest.fn();

describe('shows documents', () => {
  afterEach(cleanup);
  it('show client documents on card', () => {
    const { container } = render(
      <CardClientDownloads
        documents={[
          {
            canBeDeleted: false,
            compressed: false,
            customerId: 1,
            description: 'description',
            docCode: 'docCode',
            documentBinaryString: 'documentBinaryString',
            formId: 1,
            id: 1,
            mimeType: 'mimeType',
            typeCatagory: 'typeCatagory',
            typeId: 1,
            uploadedOn: 'uploadedOn',
          },
        ]}
        documentUrl={
          'https://brokerui-dev.mutualvision.systems/account/Customer/GetDocument/ee6a192e-73d1-4c9a-b2d7-8de883f37fc5/'
        }
        clientId={1}
        documentGridRefresh={handleRefreshDocuments}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
