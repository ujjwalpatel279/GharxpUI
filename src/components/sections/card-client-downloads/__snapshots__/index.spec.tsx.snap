// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shows documents show client documents on card 1`] = `
<div>
  <div
    class="c-card u-mb-3"
  >
    <div
      class="c-card__header"
    >
      <header
        class="c-card__header"
      >
        <div
          class="c-card__header__top"
        >
          <div
            class="c-card__header__applicant"
          >
            <h3>
              Documents
            </h3>
          </div>
        </div>
      </header>
    </div>
    <div
      class="c-card__body"
    >
      <div
        class="s-card-client__detail"
      >
        <span
          class="doc-upload__success"
        >
          <svg
            class="c-icon c-icon--primary__tick"
            height="16"
            viewBox="0 0 16 16"
            width="16"
          >
            <path
              d="M14.92 4.768c0 0.223-0.089 0.446-0.25 0.607l-7.679 7.679c-0.161 0.161-0.384 0.25-0.607 0.25s-0.446-0.089-0.607-0.25l-4.446-4.446c-0.161-0.161-0.25-0.384-0.25-0.607s0.089-0.446 0.25-0.607l1.214-1.214c0.161-0.161 0.384-0.25 0.607-0.25s0.446 0.089 0.607 0.25l2.625 2.634 5.857-5.866c0.161-0.161 0.384-0.25 0.607-0.25s0.446 0.089 0.607 0.25l1.214 1.214c0.161 0.161 0.25 0.384 0.25 0.607z"
            />
          </svg>
        </span>
        <div
          class="s-card-client__detail__content"
        >
          <span>
            description
          </span>
        </div>
        <a
          href="https://brokerui-dev.mutualvision.systems/account/Customer/GetDocument/ee6a192e-73d1-4c9a-b2d7-8de883f37fc5/1/1.undefined"
          rel="noreferrer"
          target="_blank"
        >
          <button
            class="c-btn c-btn--secondary c-btn--small c-btn--icon-left"
            type="button"
          >
            <svg
              class="c-icon"
              height="16"
              viewBox="0 0 16 16"
              width="16"
            >
              <path
                d="M7.429 1v7.313l-2.023-1.686-1.097 1.317 3.977 3.314 3.977-3.314-1.097-1.317-2.023 1.686v-7.313h-1.714zM0 10.143v2.857c0 1.105 0.895 2 2 2h12c1.105 0 2-0.895 2-2v-2.857h-1.714v2.857c0 0.158-0.128 0.286-0.286 0.286h-12c-0.158 0-0.286-0.128-0.286-0.286v-2.857h-1.714z"
              />
            </svg>
            <span>
              Download
            </span>
          </button>
        </a>
      </div>
    </div>
    <div
      class="c-card__footer "
    >
      <div
        class="modal display-none doc-upload"
      >
        <section
          class="modal-main"
        >
          <form>
            <div
              class="c-form-field"
            >
              <label
                class="c-form-field__label"
              >
                File to upload
              </label>
              <input
                accept="image/png, image/jpeg, application/pdf"
                aria-invalid="false"
                id="customerDocument"
                name="customerDocument"
                type="file"
              />
              <p
                class="c-form-field__error"
                role="alert"
              >
                Please select a file of type jpg/png/pdf with a file size less than 4MB
              </p>
            </div>
            <div
              class="c-form-field"
            >
              <label
                class="c-form-field__label"
              >
                Category
              </label>
              <fieldset>
                <legend>
                  Select a document category
                </legend>
                <div
                  class="l-flex l-flex--wrap"
                />
              </fieldset>
            </div>
            <div
              class="c-form-field"
            >
              <label
                class="c-form-field__label"
              >
                File Description
              </label>
              <input
                aria-invalid="false"
                class="c-input"
                id="fileDescription"
                name="fileDescription"
                placeholder="File Description"
                type="text"
              />
            </div>
            <p
              class="u-mb-1"
            />
            <button
              class="c-btn c-btn--secondary c-btn--small"
              type="button"
            >
              <span>
                Cancel
              </span>
            </button>
            <button
              class="c-btn c-btn--primary c-btn--small disabled upload-button"
              data-value="1"
              disabled=""
              type="submit"
            >
              <span>
                Upload
              </span>
            </button>
          </form>
        </section>
      </div>
      <footer
        class="s-card-client__detail"
      >
        <button
          class="c-btn c-btn--secondary c-btn--small c-btn--icon-left"
          type="button"
        >
          <svg
            class="c-icon"
            height="16"
            viewBox="0 0 16 16"
            width="16"
          >
            <path
              d="M7 7v-7h2v7h7v2h-7v7h-2v-7h-7v-2h7z"
            />
          </svg>
          <span>
            Add document
          </span>
        </button>
      </footer>
    </div>
  </div>
</div>
`;
